include/cuco/sentinel.cuh

```diff
@@ -31,6 +31,14 @@ struct empty_key {
    * @param v The empty key sentinel value
    */
   __host__ __device__ explicit constexpr empty_key(T v) : value{v} {}
+
```

Comment:
Might as well define a common strong type utility while you're at it. 
```
template <typename T>
struct strong_type{
__host__ __device__ explicit constexpr strong_type(T v) : value{v} {}
__host__ __device__ constexpr operator T() const noexcept {return value;}
T value;
};

template <typename T>
struct empty_key : strong_type<T>{ using strong_type::strong_type; // import ctor };

template <typename T>
struct empty_value : strong_type<T>{ using strong_type::strong_type; };
```

Reply:
Hmm, adding the base class results in errors with const vs. non-const value types:

In some examples we let the value type be deduced:
```cpp
Key constexpr empty_key_sentinel = -1;
cuco::sentinel::empty_key{empty_key_sentinel}
```

which results in the following error
`error: no instance of constructor "empty_key" matches the argument list argument types are: (const Key)`

Which opens the question: Do we want the tparam to be deducible for a strong type?

Reply:
Will `__host__ __device__ explicit constexpr strong_type(T const v) : value{v} {}` solve the issue?

Reply:
Nope, more errors -.-

```
/workspaces/cuCollections/examples/static_map/custom_type_example.cu(97): error: no instance of constructor "empty_key" matches the argument list
            argument types are: (const custom_key_type)

/workspaces/cuCollections/examples/static_map/custom_type_example.cu(98): error: no instance of constructor "empty_value" matches the argument list
            argument types are: (const custom_value_type)

/workspaces/cuCollections/examples/static_map/device_view_example.cu(139): error: no instance of constructor "empty_key" matches the argument list
            argument types are: (const Key)

/workspaces/cuCollections/examples/static_map/device_view_example.cu(140): error: no instance of constructor "empty_value" matches the argument list
            argument types are: (const Value)

2 errors detected in the compilation of "/workspaces/cuCollections/examples/static_map/custom_type_example.cu".
make[2]: *** [examples/CMakeFiles/STATIC_MAP_CUSTOM_TYPE_EXAMPLE.dir/build.make:76: examples/CMakeFiles/STATIC_MAP_CUSTOM_TYPE_EXAMPLE.dir/static_map/custom_type_example.cu.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1093: examples/CMakeFiles/STATIC_MAP_CUSTOM_TYPE_EXAMPLE.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/workspaces/cuCollections/examples/static_multimap/host_bulk_example.cu(42): error: no instance of constructor "empty_key" matches the argument list
            argument types are: (key_type)

/workspaces/cuCollections/examples/static_multimap/host_bulk_example.cu(43): error: no instance of constructor "empty_value" matches the argument list
            argument types are: (value_type)

2 errors detected in the compilation of "/workspaces/cuCollections/examples/static_map/device_view_example.cu".
/workspaces/cuCollections/examples/static_map/host_bulk_example.cu(58): error: no instance of constructor "empty_key" matches the argument list
            argument types are: (const Key)

/workspaces/cuCollections/examples/static_map/host_bulk_example.cu(59): error: no instance of constructor "empty_value" matches the argument list
            argument types are: (const Value)
```

Reply:
https://godbolt.org/z/soh8rE99z Writing the ctor instead of using `using` alias will work.

Reply:
It worked, thanks! I thought importing the super class ctor would be equivalent but apparently it's not. 

Changes have been pushed.

Reply:
Oh boo. Deduction guides aren't inherited and there's no way to inherit them :( 

https://stackoverflow.com/questions/61311297/how-to-make-argument-deduction-work-for-derived-class-which-using-base-class-con